### DISCUSSION TITLE
Compile error when using action for BNDDIR

### DISCUSSION THREAD

**Discussion by Apsoftbe** [codefori/vscode-ibmi#2120] [Q&A] [answered]
Working :
?CRTPGM PGM(&OPENLIB/&OPENMBR) MODULE(*PGM) ENTMOD(*FIRST) BNDDIR(LIBHTTP/HTTPAPI) ACTGRP(*CALLER) USRPRF(*OWNER) TGTRLS(*CURRENT)

Not working : 
?CRTPGM PGM(&OPENLIB/&OPENMBR) MODULE(*PGM) ENTMOD(*FIRST)
BNDDIR(${bnddir|Binding directory|*NONE,LIBHTTP/HTTPAPI}) ACTGRP(*CALLER) USRPRF(*OWNER) TGTRLS(*CURRENT)

Compile error
![image](https://github.com/codefori/vscode-ibmi/assets/15779852/a91ec708-0ff5-4565-9147-4faf3c9ea1d9)

Doesn't seem to find the library.

 - Extension version : v2.10.1
 - IBM i OS version : V7.3

**Comment by worksofliam** [comment]
@Apsoftbe Have you confirmed your user library list is correct?

**Comment by Apsoftbe** [comment]
Liam,
Now better screenshot with beginning of the information.
![image](https://github.com/codefori/vscode-ibmi/assets/15779852/2e4c9269-c13f-4acc-bcd3-b39e4c3a32d8)
Thanks
Peter

**Reply by worksofliam** [reply]
Peter, as much as this helps - this doesn't show me anything about Code for IBM i - just that the compile failed.

**Reply by Apsoftbe** [reply]
Liam,
Can you give me a direction what kind of information I need to send related to Code for IBM i.
Thanks
Peter

**Comment by chrjorgensen** [comment]
@Apsoftbe Please verify that you do not have any newlines in your command

```
?CRTPGM PGM(&OPENLIB/&OPENMBR) MODULE(*PGM) ENTMOD(*FIRST)
BNDDIR(${bnddir|Binding directory|*NONE,LIBHTTP/HTTPAPI}) ACTGRP(*CALLER) USRPRF(*OWNER) TGTRLS(*CURRENT)
```

When I tested this it wouldn't work with the question mark in front - I had to remove the question mark to get the `BNDDIR` prompt to work. Which leads me to think you may have a newline before `BNDDIR` and makes this a second command - newlines separates commands!

**Comment by Apsoftbe** [comment]
Christian,

Situation 1 - Working removed ? in front
Get action to select
CRTPGM PGM(&OPENLIB/&OPENMBR) MODULE(*PGM) ENTMOD(*FIRST) BNDDIR(${bnddir|Bindary directory|*NONE,LIBHTTP/HTTPAPI}) ACTGRP(*CALLER) USRPRF(*OWNER) TGTRLS(*CURRENT)

Situation 2 - Not working added ? in front
Don't get action to select
Command not properly working
?CRTPGM PGM(&OPENLIB/&OPENMBR) MODULE(*PGM) ENTMOD(*FIRST) BNDDIR(${bnddir|Bindary directory|*NONE,LIBHTTP/HTTPAPI}) ACTGRP(*CALLER) USRPRF(*OWNER) TGTRLS(*CURRENT)
![image](https://github.com/codefori/vscode-ibmi/assets/15779852/7ff7dbce-c4d6-46e3-9b7d-8e9905614c8f)

Situation 3 - Not working BNDDIR on second line
Get action to select but compile isn't finding library because BNDDIR is on new line
?CRTPGM PGM(&OPENLIB/&OPENMBR) MODULE(*PGM) ENTMOD(*FIRST) 
BNDDIR(${bnddir|Bindary directory|*NONE,LIBHTTP/HTTPAPI}) ACTGRP(*CALLER) USRPRF(*OWNER) TGTRLS(*CURRENT)
![image](https://github.com/codefori/vscode-ibmi/assets/15779852/dbb5ee27-c34e-4f40-ba8e-5abdf24a0c21)

Thanks
Peter

**Reply by sebjulliand** [reply, marked as answer]
Got it! You cannot mix custom inputs (e.g. `${bnddir|Bindary directory|*NONE,LIBHTTP/HTTPAPI}`) and prompting (when the commands starts with a `?`). They are mutually exclusive and custom inputs take precedence over prompting; which explains why Situation 2 is not working. Situation 3 is invalid because you split the command. Each new line of an action command must be a full command.

Situation 1 is the way to go for your use case.

**Comment by Apsoftbe** [comment]
Sébastian,
I will write it down that i can't use displaying command and prompt for values at the same time.
It is already in the documentation but i probably misread or missed it. See Developing/Actions/Execution/Prompted (first line).
Thanks
Peter
